src/ServiceInsight/ServiceControl/DefaultServiceControl.cs

```diff
@@ -145,7 +145,11 @@ public Task<PagedResult<StoredMessage>> GetAuditMessages(string link)
             var rootUrls = await GetRootUrls().ConfigureAwait(false);
 
             var endpointsUrl = rootUrls?.KnownEndpointsUrl ?? rootUrls?.EndpointsUrl ?? DefaultEndpointsEndpoint;
-            endpointsUrl = endpointsUrl.Replace(connection.Url, string.Empty);
+            if (Uri.TryCreate(endpointsUrl, UriKind.RelativeOrAbsolute, out var endpointsUri) &&
```

Comment:
So if we can't create the URIs we will use whatever that was provided. Can we envision any scenarios where that would fail?

Reply:
In that case, both the SI connection URL and the SC URL are garbage, and things should rightly blow up.

Reply:
Note that there is no safeguard against that in 1.9.1, 1.10.0 or after this change. The queries will fail in all cases and the log output is all we have to go on.